llmfoundry/data/text_data.py

```diff
@@ -193,11 +194,12 @@ def __init__(
                 '`bos_token_id` if sequences start with a BOS token.'
             )
 
-        self.split_token_id = eos_token_id
-        self.bos_mode = False
         if eos_token_id is None:
-            self.split_token_id = bos_token_id
+            self.split_token_id = cast(int, bos_token_id)
```

Comment:
Why use `cast(int, bos_token_id)` instead of just `int(bos_token_id)`?

Reply:
The intent here is to force the type given the safeguards above (thus using `cast`), not try to convert to an `int`